enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,320 @@
+type DeviceMechanicalProperty string
+
+const (
+	// The mechanical properties of the devices
+	// Rotational refers to magnetic disks
+	Rotational DeviceMechanicalProperty = "Rotational"
+	// NonRotational refers to ssds
+	NonRotational DeviceMechanicalProperty = "NonRotational"
+)
+
+type DeviceDiscoveryPolicyType string
+
+const (
+	// The DeviceDiscoveryPolicies that will be supported by the LSO.
+	// These Discovery policies will based on lsblk's type output
+	// Disk represents a device-type of disk
+	Disk DeviceDiscoveryPolicyType = "disk"
+	// Part represents a device-type of partion
+	Part DeviceDiscoveryPolicyType = "part"
+)
+
+type DeviceInclusionSpec struct {
+	// DeviceTypes that should be used for automatic detection. This would be one of the types supported
+	// by the local-storage operator. Currently, the supported types are: disk, part
+	// If the list is empty the default value will be `[disk]`.
+	DeviceTypes []DeviceDiscoveryPolicyType `json:"deviceType"`
+
+	// DeviceMechanicalProperty denotes whether Rotational or NonRotational disks should be used.
+	// by default, it selects both
+	// +optional
+	DeviceMechanicalProperty DeviceMechanicalProperty `json:"deviceMechanicalProperty"`
+
+	// MinSize is the minimum size of the device which needs to be included
+	// +optional
+	MinSize resource.Quantity `json:"minSize"`
+
+	// MaxSize is the maximum size of the device which needs to be included
+	// +optional
+	MaxSize resource.Quantity `json:"maxSize"`
+
+	// Models is a list of device models. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Models []string `json:"models"`
+
+	// Vendors is a list of device vendors. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Vendors []string `json:"vendors"`
+}
+
+type LocalVolumeGroupPhase string
+
+const (
+	DiscoveringPhase LocalVolumeGroupPhase = "Discovering"
+	FailedPhase      LocalVolumeGroupPhase = "Failed"
+	DiscoveredPhase  Phase = "Discovered"
+	ProvisionedPhase LocalVolumeGroupPhase = "Discovered"
+)
+
+type LocalVolumeGroupStatus struct {
+	Phase                       LocalVolumeGroupPhase `json:"phase,omitempty"`
+	TotalDiscoveredDeviceCount  *int32                `json:"totaldiscovereddevicecount,omitempty"`
+	TotalProvisionedDeviceCount *int32                `json:"totalprovisioneddevicecount,omitempty"`
+	TimeStramp                  `json:"phase,omitempty"`
+}
+```
+### Workflow of LocalPV creation via LocalVolumeGroup CR
+- Once the `LocalVolumeGroup` is created, the controller will:
+  - configure the local-static-provisioner to make a new StorageClass based on certain directories on the selected nodes.
+  - assign diskmaker daemons to the selected nodes.
+- The diskmaker daemon will find devices that match the disovery policy and symlink them into the directory that the local-static-provisioner is watching.
```

Comment:
It must be clearly described what happens when;

* Two `LocalVolumeGroup` CRs target a single node
* Two `LovalVolumeGroup` CRs target a single node and match a single local volume.
* `LocalVolume` and `LocalVolumeGroup` target a single node (and a single volume).

Reply:
added